router/h2/src/main/scala/io/buoyant/router/h2/ClassifierFilter.scala

```diff
@@ -11,40 +14,71 @@ object ClassifierFilter {
   val SuccessClassHeader = "l5d-success-class"
 
   def module: Stackable[ServiceFactory[Request, Response]] =
-    new Stack.Module1[param.ResponseClassifier, ServiceFactory[Request, Response]] {
-      val role = ClassifierFilter.role
-      val description = "Sets the response classification into a header"
-      def make(classifierP: param.ResponseClassifier, next: ServiceFactory[Request, Response]) = {
-        val param.ResponseClassifier(classifier) = classifierP
+    new Stack.Module1[param.H2Classifier, ServiceFactory[Request, Response]] {
+      override val role: Stack.Role = ClassifierFilter.role
+      override val description = "Sets the stream classification into a header"
+      override def make(
+        classifierP: param.H2Classifier,
+        next: ServiceFactory[Request, Response]
+      ): ServiceFactory[Request, Response] = {
+        val param.H2Classifier(classifier) = classifierP
         new ClassifierFilter(classifier).andThen(next)
       }
     }
 
-  val successClassClassifier: ResponseClassifier = {
-    case ReqRep(req, Return(rep: Response)) if rep.headers.contains(SuccessClassHeader) =>
-      val success = rep.headers.get(SuccessClassHeader)
-        .flatMap { value =>
-          Try(value.toDouble).toOption
-        }.getOrElse(0.0)
-      if (success > 0.0)
-        ResponseClass.Successful(success)
-      else
-        ResponseClass.Failed(false)
+  private[this] object ResponseSuccessClass {
+    @inline def unapply(headers: Headers): Option[ResponseClass] =
+      headers.get(SuccessClassHeader).map { value =>
+        val success = value.toDouble
```

Comment:
I was considering that, just wasn't sure what the correct way to handle the error was in this case.